### DISCUSSION TITLE
Can rust part be use as dynamic loading during the runtime?

### DISCUSSION THREAD

**Discussion by michalss** [fzyzcjy/flutter_rust_bridge#2386] [General]
Hello,

Can this package work like, rust part as dynamic plugin. Means load them in the flutter app in the runtime ? Im missing such a functionality in flutter. Simple make plugins where i can load them from flutter in runtime.. I mean make, rust part be done separately from main app and load with flutter and not build with main application together?

Thx

**Comment by fzyzcjy** [comment]
Are you targeting at ios or other platforms? iirc ios has a strict requirement that you cannot download a binary that is executable, so I guess it may not be possible for ios; but can be good for android etc.

But if you are happy to compile Rust to WASM, it is possible to have it in ios (and other platforms). An example is https://pub.dev/packages/rfw#integrating-with-scripting-language-runtimes.

(There may need some change to frb in order to support these.)

**Reply by michalss** [reply]
mosty do care about desktops and Android... iOS not important ...

**Reply by fzyzcjy** [reply]
Then that's great! As long as you can download your on-the-fly `.so` file somewhere, then you can point frb's loader to that file, and things should work.

**Reply by michalss** [reply]
so can it be done with rust then ? is there any sample how to pls ?

**Reply by fzyzcjy** [reply]
I guess you can do something like:

1. Compile your rust code into a `.so` file. Frb uses cargokit to integrate the compilation process into flutter automatically, so here you need to do it manually. Some doc in https://cjycode.com/flutter_rust_bridge/manual/integrate/overview may help.
2. Somehow download your `so` file on the fly.
3. When `RustLib.init`, point to your downloaded so file.

**Reply by michalss** [reply]
thx a lot, only not understandable part is i dont know how to complile own so file manually yet...

**Reply by fzyzcjy** [reply]
You are welcome!

>  i dont know how to complile own so file manually yet

Indeed it is nothing but some `cargo build` under the hood, just with correct flags. Maybe check the doc mentioned, or maybe run a normal frb project and see what command cargokit executes indeed, and use that as a reference.
